File: templates/java/team_source/Strategy.java
import java.util.random.RandomGenerator;

public final class Strategy {
    private Strategy() {}

    /** Returns one legal move using only the wrapper-provided random stream. */
    public static String chooseMove(
        int turn,
        String myHistory,
        String opponentHistory,
        RandomGenerator rng
    ) {
        String[] moves = {"R", "P", "S"};
        return moves[rng.nextInt(moves.length)];
    }
}

File: templates/kotlin/team_source/Strategy.kt
import java.util.random.RandomGenerator

object Strategy {
    /** Returns one legal move using only the wrapper-provided random stream. */
    fun chooseMove(
        turn: Int,
        myHistory: String,
        opponentHistory: String,
        rng: RandomGenerator
    ): String {
        val moves = arrayOf("R", "P", "S")
        return moves[rng.nextInt(moves.size)]
    }
}
